src/App.tsx

```diff
@@ -1,31 +1,114 @@
+/* eslint-disable no-case-declarations */
+/* eslint-disable react/no-did-update-set-state */
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 interface State {
   selectedUserId: number;
+  todos: Todo[];
+  filtredTodos: Todo[];
+  filterInputValue: string;
+  filterBy: string;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
     selectedUserId: 0,
+    todos: [],
+    filtredTodos: [],
+    filterInputValue: '',
+    filterBy: '',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({
+      todos,
+      filtredTodos: todos,
+    });
+  }
+
+  componentDidUpdate(_: unknown, prevState: State) {
+    const { filterInputValue, todos, filterBy } = this.state;
+
+    if (prevState.filterInputValue !== filterInputValue) {
+      const filtredTodos = todos.filter(todo => todo.title.includes(filterInputValue));
+
+      this.setState({ filtredTodos });
+    }
+
+    if (prevState.filterBy !== filterBy) {
+      switch (filterBy) {
+        case 'all':
+          this.setState({ filtredTodos: todos });
+          break;
+
+        case 'not completed':
+          const filtredByNotCompleted = todos.filter(todo => todo.completed === false);
+
+          this.setState({ filtredTodos: filtredByNotCompleted });
+          break;
+
+        case 'completed':
+          const filtredByCompleted = todos.filter(todo => todo.completed === true);
+
+          this.setState({ filtredTodos: filtredByCompleted });
+          break;
+
+        default:
+          return null;
+      }
+    }
```

Comment:
call filtering after clicking on option on select, no need to do it in compomentDidUpdate